sentinelsat/sentinel.py

```diff
@@ -806,30 +885,43 @@ def _md5_compare(self, file_path, checksum, block_size=2 ** 13):
                     progress.update(len(block_data))
             return md5.hexdigest().lower() == checksum.lower()
 
-    def _download(self, url, path, session, file_size):
-        headers = {}
-        continuing = exists(path)
-        if continuing:
-            already_downloaded_bytes = getsize(path)
-            headers = {'Range': 'bytes={}-'.format(already_downloaded_bytes)}
-        else:
-            already_downloaded_bytes = 0
-        downloaded_bytes = 0
-        with closing(session.get(url, stream=True, auth=session.auth,
-                                 headers=headers, timeout=self.timeout)) as r, \
-                closing(self._tqdm(desc="Downloading", total=file_size, unit="B",
-                                   unit_scale=True, initial=already_downloaded_bytes)) as progress:
-            _check_scihub_response(r, test_json=False)
-            chunk_size = 2 ** 20  # download in 1 MB chunks
-            mode = 'ab' if continuing else 'wb'
-            with open(path, mode) as f:
-                for chunk in r.iter_content(chunk_size=chunk_size):
-                    if chunk:  # filter out keep-alive new chunks
-                        f.write(chunk)
-                        progress.update(len(chunk))
-                        downloaded_bytes += len(chunk)
-            # Return the number of bytes downloaded
-            return downloaded_bytes
+    def _download(self, url, path, session, file_size, product_info=None):
```

Comment:
You don't need the optional `product_info`. The function could also just return / append whatever a list of the `url`s that were passed in. In any case, the tracking of failure / success should happen outside the downloading function.

Reply:
yes sure, I remember that next time!